plugin-locationcomponent/src/main/java/com/mapbox/maps/plugin/locationcomponent/LocationPuckManager.kt

```diff
@@ -49,20 +49,22 @@ internal class LocationPuckManager(
     }
 
   fun initialize(style: StyleInterface) {
-    animationManager.setUpdateListeners(onLocationUpdated, onBearingUpdated)
-    animationManager.setLocationLayerRenderer(locationLayerRenderer)
-    animationManager.applyPulsingAnimationSettings(settings)
-    locationLayerRenderer.addLayers(positionManager)
-    lastLocation?.let {
-      updateCurrentPosition(it)
-    }
-    updateCurrentBearing(lastBearing)
-    locationLayerRenderer.initializeComponents(style)
-    styleScaling(settings)
-    if (lastLocation != null && settings.enabled) {
-      show()
-    } else {
-      hide()
+    if (!locationLayerRenderer.isRendererInitialised()) {
```

Comment:
what about else branch, perhaps any log entry needed?

Reply:
Not sure, with the current arch of settings applied (async getStyle returning) that's a valid and expected case that this can happen multiple times.